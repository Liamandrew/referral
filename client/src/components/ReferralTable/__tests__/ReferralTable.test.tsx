import React from "react";
import { render } from "@testing-library/react";
import { ReferralTable } from "../ReferralTable";

describe("<ReferralTable>", () => {
    test("should match the snapshot", () => {
        const { container } = render(
            <ReferralTable
                referrals={[
                    {
                        id: 3,
                        email: "email",
                        givenName: "givenName",
                        surname: "surname",
                        phoneNumber: "phoneNumber",
                    },
                ]}
                onReferralDeleteClick={jest.fn}
                onReferralEditClick={jest.fn}
            />
        );

        expect(container).toMatchSnapshot();
    });
});
