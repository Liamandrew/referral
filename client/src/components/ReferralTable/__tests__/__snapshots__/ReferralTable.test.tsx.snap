// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReferralTable> should match the snapshot 1`] = `
<div>
  <div
    class="MuiTableContainer-root container"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root tableHeadCell MuiTableCell-head"
            scope="col"
          >
            Given Name
          </th>
          <th
            class="MuiTableCell-root tableHeadCell MuiTableCell-head"
            scope="col"
          >
            Surname
          </th>
          <th
            class="MuiTableCell-root tableHeadCell MuiTableCell-head"
            scope="col"
          >
            Email
          </th>
          <th
            class="MuiTableCell-root tableHeadCell MuiTableCell-head"
            scope="col"
          >
            Phone
          </th>
          <th
            class="MuiTableCell-root tableHeadCell MuiTableCell-head"
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root tableBodyCell MuiTableCell-body"
          >
            givenName
          </td>
          <td
            class="MuiTableCell-root tableBodyCell MuiTableCell-body"
          >
            surname
          </td>
          <td
            class="MuiTableCell-root tableBodyCell MuiTableCell-body"
          >
            email
          </td>
          <td
            class="MuiTableCell-root tableBodyCell MuiTableCell-body"
          >
            phoneNumber
          </td>
          <td
            class="MuiTableCell-root actionBodyCell MuiTableCell-body"
          >
            <button
              class="iconButton"
            >
              <svg>
                create-24px.svg
              </svg>
            </button>
            <button
              class="iconButton"
            >
              <svg>
                delete-24px.svg
              </svg>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
